### DISCUSSION TITLE
RPC passing a param of type array?

### DISCUSSION THREAD

**Discussion by malalecherocks** [supabase/supabase#8052] [Questions] [answered]
Hi, I want to delete a bunch of rows passing an array of ids, something like this:

```
let arrayOfIds = ['234','443']
const { data, error } = await supabase
  .rpc('delete_queue', { _ids: arrayOfIds })
```

And my function doesn't seem to work, probably something small missing here:

```
-- Parameters need to be prefixed because the name clashes with `om`'s columns
CREATE FUNCTION delete_queue(_ids text[]) RETURNS bool AS $$
DELETE FROM wa_queue_task WHERE id_sec IN  _ids;
$$ LANGUAGE sql SECURITY DEFINER;
-- Function is owned by postgres which bypasses RLS
```

What am I doing wrong?

By the way I'm doing a RPC because I tried to use supabase javascript sdk to delete an array of 1,000 Ids... and it throwed me an error of URI too long.

**Comment by GaryAustin1** [comment, marked as answer]
Try id_sec = ANY (_ids)  for an array versus list.

**Comment by malalecherocks** [comment]
yeah Gary, I spent 2 hours searching and you with just few words solved the problem, thanks! here it is the solution for everyone:

```
create function delete_wa_queue_task(_ids int[]) returns void as $$
   DELETE FROM wa_queue_task WHERE id_sec = ANY (_ids);
$$ language sql;
select delete_wa_queue_task(array [23,4,234]);
```

Now I'm going to try the RPC sending an array of 1000 items and see if it works...

**Reply by malalecherocks** [reply]
yeah it works sending an array of 1000 items, while the ordinary:

```
const { data: dataDelete, error: errorDelete } = await supabase
.from('wa_queue_task')
.delete()
.in('id_sec', aIdQueueSent);
```
sends an error of URI too long
